tests/lib/testing_classes.py

```diff
@@ -365,26 +377,42 @@ def test_folder_structure_same(self, dbcon, publish_finished,
         expected_dir_base = os.path.join(download_test_data,
                                          "expected")
 
-        print("Comparing published:'{}' : expected:'{}'".format(
-            published_dir_base, expected_dir_base))
-        published = set(f.replace(published_dir_base, '') for f in
-                        glob.glob(published_dir_base + "\\**", recursive=True)
-                        if f != published_dir_base and os.path.exists(f))
-        expected = set(f.replace(expected_dir_base, '') for f in
-                       glob.glob(expected_dir_base + "\\**", recursive=True)
-                       if f != expected_dir_base and os.path.exists(f))
+        print(
+            "Comparing published:'{}' : expected:'{}'".format(
+                published_dir_base, expected_dir_base
+            )
+        )
+
+        def get_files(dir_base):
+            result = set()
+
+            for f in glob.glob(dir_base + "\\**", recursive=True):
```

Comment:
Since you are basically globbing all recursively why not use `os.walk`? (or `os.scandir` is that's still a thing that it's faster)

Reply:
Is using `os.walk` for speed?

Reply:
I thought it might also improve readability - but I also suspect it's faster as it doesn't need to do the string comparisons, etc + you don't need the additional `isdir` check.

Speed-wise I don't think you'll really notice anything for the amount of files we're likely processing here.

Reply:
Would be interested in how big performance gains it would produce. It looks more like fixing plastic pollution by banning of plastic straws ;).
But feel free to modify it, irrelevant to this PR though.

Reply:
Agree - was just wondering initially why `glob` was used.

Reply:
Not sure, I think I was getting frustrated by seemingly simple use case that I expected would have been already solved zillion times by some built-in or third party lib.